store/postgres/src/chain_store.rs

```
@@ -1413,6 +1450,74 @@ impl ChainStoreTrait for ChainStore {
             .block_number(&conn, hash)?
             .map(|number| (self.chain.clone(), number)))
     }
+
+    /// Tries to obtain transaction statuses for a given block range.
+    ///
+    /// According to EIP-658, there are two ways of checking if a transaction failed:
+    /// 1. by checking if it ran out of gas.
+    /// 2. by looking at its receipt "status" boolean field, which may be absent for blocks before
+    ///    Byzantium fork.
+    ///
+    /// This function will only compare gas exhaustion for transactions when their respective
+    /// receipts contain no information about their status.
+    fn transaction_statuses_in_block_range(
+        &self,
+        block_range: &Range<BlockNumber>,
+    ) -> Result<HashMap<H256, bool>, StoreError> {
+        let conn = self.get_conn()?;
+        let receipts = self.storage.find_transaction_receipts_for_block_range(
+            &conn,
+            &self.chain,
+            &block_range,
+        )?;
+
+        let mut statuses = HashMap::new();
+        let mut pending = HashMap::new();
+        for receipt in receipts.into_iter() {
+            match (receipt.status, receipt.gas_used) {
```

If `gas_used` is `None` this should return an error.